tools/import-pronom-ids

```diff
@@ -0,0 +1,157 @@
+        print("    FormatVersion = apps.get_model('fpr', 'FormatVersion')", file=output_file)
+        print("    IDRule = apps.get_model('fpr', 'IDRule')", file=output_file)
+        print(file=output_file)
+
+    for format in formats.getroot():
+        puid = format.find('.puid').text
+        # fmt/111 is "OLE2 Compound Document Format".
+        # Many Windows file formats that aren't otherwise IDed will
+        # be identified as this. We choose not to import this one
+        # ID in order to allow Archivematica to recognize those files
+        # as unidentified instead of returning something misleading.
+        if puid == "fmt/111":
+            continue
+        # If a FormatVersion with this PUID already exists,
+        # we don't want to do anything.
+        try:
+            FormatVersion.objects.get(pronom_id=puid)
+            continue
+        except FormatVersion.DoesNotExist:
+            pass
+
+        new_format = PronomFormat(format)
+        print('Importing', new_format.version_name, file=sys.stderr)
+        parent_format = archivematica_formats.get(new_format.format_name)
+        if not parent_format:
+            try:
+                parent_format = Format.objects.get(description=new_format.format_name)
+            except Format.DoesNotExist:
+                parent_format = Format(
+                    description=new_format.format_name,
+                    group=unknown_format_group,
+                    uuid=str(uuid.uuid4())
+                )
+                migration = '''    Format.objects.create(description="{}", group_id="{}", uuid="{}")'''.format(new_format.format_name, unknown_format_group.uuid, parent_format.uuid)
+                sql = save_object(parent_format)
+                choose_output(output_format, output_file, sql, migration)
+                archivematica_formats[new_format.format_name] = parent_format
+
+        format_version = FormatVersion(
+            format=parent_format,
+            pronom_id=new_format.puid,
+            description=new_format.version_name,
+            version=new_format.version,
+            uuid=str(uuid.uuid4())
+        )
+        sql = save_object(format_version)
+        migration = '''    FormatVersion.objects.create(format_id="{}", pronom_id="{}", description="{}", version="{}", uuid="{}")'''.format(parent_format.uuid, new_format.puid, new_format.version_name, new_format.version, format_version.uuid)
+        choose_output(output_format, output_file, sql, migration)
+
+        # If an extension is listed, set up a new IDRule so that
+        # the `Files by Extension` IDCommand can find them.
+        if new_format.extension:
+            # First check to see if a rule already exists: if it does, delete it
+            # IDRules can only map an extension to a single format and we don't know which extension to map to.
+            try:
+                rule = IDRule.objects.get(command_output=new_format.extension)
+                rule.delete()
```

Comment:
When the IDRule is deleted, I believe we will also want the migration/sql file to contain that action as well, i.e., ``DELETE FROM `fpr_idrule` WHERE `fpr_idrule`.`id` IN (...) ;`` in the .sql output and something like `IDRule.objects.filter(command_output="...").delete()` in the migrations output. Adding something like the following three lines would do it:

    sql = connection.queries[-1]['sql']
    migration = '''    IDRule.objects.filter(command_output="{}").delete()'''.format(new_format.extension)
    choose_output(output_format, output_file, sql, migration)

If we don't do something like the above, then loading the generated .sql file will have a different effect than running the `import-pronom-ids` command. In particular, there will be more fpr_idrule rows in the former case than in the latter.